### ISSUE TITLE
Feature: pass unknown escape sequences through

### ISSUE THREAD

**Issue by a-kenji** [zellij-org/zellij#892]
> Regarding alacritty url recognition, tmux with mouse mode enabled has the same problem.

tmux swallows all escape sequences it does not know for whatever reason. I had the hope zellij does not do that.

_Originally posted by @SuperSandro2000 in https://github.com/zellij-org/zellij/issues/627#issuecomment-898263334_

Maybe something that is possible post - termion?
Is it something that is desirable?

**Comment by imsnif** [comment]
So, there are two things here:
1. Escape sequences that we get through STDIN - we pass them through as is unless we interpret them as a key we need to react to or something like bracketed paste. I'm not aware of any case we don't do this - if there is one, it's definitely a bug.
2. Escape sequences that we get through the shell or whatever is running in the terminal (vim, htop, etc.) Here we have no choice but to swallow these, because for all intents and purposes we are the terminal emulator for those tools. We can't let them do stuff like move the cursor outside their pane for example. Even in the example in the thread, if we just pass the OSC 8 ANSI escape through, it will be lost when we scroll inside the pane, for example (also other stuff like it not wrapping properly and being broken on pane line breaks). The solution here is to support as many features as possible, we unfortunately cannot rely on the user's terminal emulator in this case. :/

Or... maybe I misunderstood something here and we're talking about something else?

**Comment by a-kenji** [comment]
Thanks @imsnif for the answer!

> Or... maybe I misunderstood something here and we're talking about something else?

Maybe? I am not sure. It seems reasonable. I just saw this comment and wanted to have it addressed, one way or the other.

@SuperSandro2000, do you mind giving more context here, if you have time?

**Comment by SuperSandro2000** [comment]
> if we just pass the OSC 8 ANSI escape through, it will be lost when we scroll inside the pane

I am pretty sure there is a way to support those escape sequences with less downsides than not supporting them at all. For example the link could only be rendered when the entire text is visible or only on the parts that are visible.

**Comment by SuperSandro2000** [comment]
How about OSC 7 and OSC 133 https://wezfurlong.org/wezterm/shell-integration.html ?

**Comment by imsnif** [comment]
I'd be very happy to support them, for sure. But we also can't just pass them through as is. 

An example of OSC 7 to illustrate:
1. One pane sends an OSC 7 to Zellij telling it the working directory is "/home/pane1". Zellij passes it to the terminal emulator which logs the current directory as "/home/pane1".
2. The user opens another pane, which sends an OSC 7 to Zellij telling it the working directory is "/home/pane2", Zellij forwards it to the terminal emulator  which logs the current directory as "/home/pane2"
3. The user changes focus back to pane 1, no further signals are sent because neither of the panes know anything changed. The terminal emulator thinks the working directory is "/home/pane2" while it actually is "/home/pane1".

Another example for OSC 133:
1. We have two panes open that are vertically split (side by side). The pane on the left sends an `OSC 133;A` to illustrate the start of a command. Zellij forwards it as is to the terminal emulator.
2. The pane on the left writes a few lines to the terminal which are rendered in the left pane and then terminates the command with an `OSC 133;B`.
3. Since there's a pane on the right, the terminal emulator thinks the command also includes those lines on the pane to the right, because it doesn't know what panes are and that's what we told it to do.

In short - we have to keep track of these things. These are two examples, but there are loads of others. This is the reason we swallow these codes - because otherwise the terminal emulator won't do what the user or the command are expecting it to.

As I said - I'm totally up to supporting all of these codes. It's some work, but not more than we've already done.

**Comment by a-kenji** [comment]
@SuperSandro2000 
Thank you for that context!
I'll open separate issues with that information,
I think it would be awesome if we could utilize those sequences well.
